### PR TITLE
feat(ui): adds initialCounterValue prop

### PR DESCRIPTION
**Pull request by scefali**
This PR adds an `initialCounterValue` prop to the List component. This prop sets the initial counter value for list items. This is needed for instructions that are rendered in multiple views. PR using this: https://github.com/getsentry/sentry/pull/23298

### CODE REVIEW

File: src/sentry/static/sentry/app/components/list/utils.tsx [hunk 1]
```diff
@@ -24,7 +24,15 @@ const bulletStyle = (theme: Theme) => css`
...
 
-const numericStyle = (theme: Theme, isSolid = false) => css`
+type Options = {
+  isSolid?: boolean;
+  initialCounterValue?: number;
+};
+
+const numericStyle = (
+  theme: Theme,
+  {isSolid = false, initialCounterValue = 0}: Options
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/list/utils.tsx:34]: should it default to 0 or 1?

File: src/sentry/static/sentry/app/components/list/utils.tsx [hunk 2]
```diff
@@ -24,7 +24,15 @@ const bulletStyle = (theme: Theme) => css`
   }
 `;
 
-const numericStyle = (theme: Theme, isSolid = false) => css`
+type Options = {
+  isSolid?: boolean;
+  initialCounterValue?: number;
+};
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/list/utils.tsx:30]: Could you drop a couple doc comments above these?

**Review by evanpurkhiser — APPROVED** [review]

File: src/sentry/static/sentry/app/components/list/utils.tsx (hunk 1 above)
Comment (scefali) [reply src/sentry/static/sentry/app/components/list/utils.tsx:34]: @EvanPurkhiser Good question. Starting at `0` means the first step you see is actually `1`.

Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/components/list/utils.tsx:34]: maybe we can do some math on this value then so it matches what you see

**Pull request merged**